Support history v5

Hypnosphi opened the pull request:
Listener signature has changed from `(location, action) => {...}` to `({location, action}) => {...}`

Hypnosphi committed:
Support history v5

Treora commented:
Thanks for this PR; it’s good to keep up with the history module indeed.

> Listener signature has changed from `(listener, action) => {...}` to `({listener, action}) => {...}`

I wonder, how does that relate to the single change you committed?

        // Support history v5
        if (location.location != null) {
            location = location.location;
        }

Also I don’t see significant breaking changes on the module’s [release page](https://github.com/ReactTraining/history/releases/tag/v5.0.0). Could you explain the need?

Hypnosphi commented:
Unfortunately, there's a lot of unlisted breaking changes: https://github.com/ReactTraining/history/issues/811

> history.listen() API changed. The callbacks are now called with an an object of shape Update instead of Location

> I wonder, how does that relate to the single change you committed

Sorry for typo, it should be `location` not `listener`. In v5, the first argument is not `location`, but an object of shape `{location, action}`. In that case, I extract `.location` property from it.

Treora committed:
use strict equality test

Using `!= null` is a bit confusing: it took me a moment to realise that this would also be true if the value is `undefined`, which (I presume) is the value we should test for. Using non-strict equal tests is often discouraged for such reasons.

Treora commented:
> Sorry for typo, it should be `location` not `listener`. In v5, the first argument is not `location`, but an object of shape `{location, action}`. In that case, I extract `.location` property from it.

Ok, makes sense.

I made one tweak, as using `!= null` is a bit confusing: it took me a moment to realise that this would also be ~~true~~ *(I meant false, of course)* if the value is `undefined`, which (I presume) is the value we should test for. Using non-strict equal tests is often discouraged for such reasons. Give a shout if we should actually test for `null` too.

Treora commented:
Published on npmjs.com as redux-query-sync@0.1.10
